### PR TITLE
AO, DS

### PR DESCRIPTION
**Pull request by mrogowski**

### CODE REVIEW

File: src/binding/petsc4py/src/petsc4py/PETSc/AO.pyx [hunk 1]
```diff
@@ -9,23 +9,76 @@ class AOType(object):
...
+            The application ordering.
+        petsc
+            Another ordering (may be `None` to indicate the natural ordering,
+            that is 0, 1, 2, 3, ...).
+        comm
+            The MPI communicator that is to share the application ordering.
+
+        See also
+        --------
+        petsc.AOCreateBasicIS, petsc.AOCreateBasic, createMemoryScalable,
```
Comment (stefanozampini) [inline src/binding/petsc4py/src/petsc4py/PETSc/AO.pyx:78]: python links go first, then C

File: src/binding/petsc4py/src/petsc4py/PETSc/DS.pyx [hunk 2]
```diff
@@ -16,83 +17,412 @@ cdef class DS(Object):
         self.ds  = NULL
 
     def view(self, Viewer viewer=None):
+        """View a PetscDS
+
+        Parameters
+        ----------
+        prob - the PetscDS object to view
```
Comment (stefanozampini) [inline src/binding/petsc4py/src/petsc4py/PETSc/DS.pyx:24]: not compliant with numpydoc

Comment (mrogowski) [reply src/binding/petsc4py/src/petsc4py/PETSc/DS.pyx]: this is a draft PR, but you can review DS and AO for now

**Review by nocollier — COMMENTED** [review]
Minor changes, looking good.

**Comment by mrogowski** [comment]
Thanks for the review @nocollier, we're ready to merge that once anyone approves

@dalcinl

**Review by nocollier — APPROVED** [review]

**Pull request merged**